examples/mysql/README.md

```diff
@@ -0,0 +1,88 @@
+MySQL filter example
+====================
+
+In this example, we show how the [MySQL
+filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/network_filters/mysql_proxy_filter)
+can be used with the Envoy proxy. The Envoy proxy [configuration](./envoy.yaml)
+includes a MySQL filter that parses queries and collects MySQL-specific
+metrics.
+
+## Usage
+
+Build and run the containers:
+
+```console
+$ docker-compose build && docker-compose up
+```
+
+Use `mysql` to issue some commands and verify they are routed via Envoy. Note
+that the current implementation of the protocol filter is known to have
+problems with versions of the MySQL protocol post version 5.5.
+
+```console
+$ docker run --rm -it --network envoymesh mysql:5.5 mysql -h envoy -P 1999 -u root
+... snip ...
+
+mysql> CREATE DATABASE test;
+Query OK, 1 row affected (0.00 sec)
+
+mysql> USE test;
+Database changed
+
+mysql> CREATE TABLE test ( text VARCHAR(255) );
+Query OK, 0 rows affected (0.02 sec)
+
+mysql> SELECT COUNT(*) FROM test;
++----------+
+| COUNT(*) |
++----------+
+|        0 |
++----------+
+1 row in set (0.00 sec)
+
+mysql> INSERT INTO test VALUES ('hello, world!');
+Query OK, 1 row affected (0.01 sec)
+
+mysql> SELECT COUNT(*) FROM test;
++----------+
+| COUNT(*) |
++----------+
+|        1 |
++----------+
+1 row in set (0.00 sec)
+
+mysql> exit
+Bye
+```
+
+Check that the egress stats were updated:
+
+```console
+$ curl -s http://localhost:8001/stats?filter=egress_mysql
+mysql.egress_mysql.auth_switch_request: 0
+mysql.egress_mysql.decoder_errors: 0
+mysql.egress_mysql.login_attempts: 1
+mysql.egress_mysql.login_failures: 0
+mysql.egress_mysql.protocol_errors: 0
+mysql.egress_mysql.queries_parse_error: 0
+mysql.egress_mysql.queries_parsed: 7
+mysql.egress_mysql.sessions: 1
+mysql.egress_mysql.upgraded_to_ssl: 0
```

Comment:
We collect lot more than this. Unfortunately its all in dynamic metadata. If we could print it in the access logs, then that would rock.. it includes tables accessed, and operations on the table

Reply:
+1

Reply:
Yeah, that's actually what led me here in the first place. I made #6233 to track this improvement / feature request.